### PR TITLE
[sinon-express-mock] express compatibility

### PR DESCRIPTION
**Pull request by tlaziuk**
* improve tests and make the typings compatible with `express`
* add generic options param

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/danawoodman/sinon-express-mock/blob/master/src/index.js>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by jpchip** [comment]
I will test and likely approve these changes on Monday.

**Comment by tlaziuk** [comment]
@jpchip thanks

File: types/sinon-express-mock/index.d.ts [hunk 1]
```diff
@@ -1,55 +1,63 @@
...
+        sendStatus: SinonStub;
+        set: SinonStub;
+        status: SinonStub;
+        type: SinonStub;
+        vary: SinonStub;
+    }
 }
 
-export function mockReq(options?: object): mockReq.MockReq;
+export function mockReq<T extends object>(options?: T): mockReq.MockReq & T & Request;
```
Comment (jpchip) [inline types/sinon-express-mock/index.d.ts]: I was wondering if there was a way to do this without having to implement a stub for EVERY property in the Request/Response object. Learned something new, thanks!

**Review by jpchip — APPROVED** [review]
LGTM

File: types/sinon-express-mock/index.d.ts (hunk 1 above)
Comment (tlaziuk) [inline types/sinon-express-mock/index.d.ts]: I don't know if this is the proper way either. Now typings "thinks" there is eg. cookies property in that mock, when there is no such property. Thankfully the cookies is any, but there are also other properties which are not included in that mock. I think you know what I want to say 😊

**Comment by tlaziuk** [comment]
done, I don't know why do those conflicts suddenly appeared

**Comment by tlaziuk** [comment]
@jpchip I think your review is needed once more 😊

**Review by jpchip — APPROVED** [review]
LGTM again

**Pull request merged**